src/rgw/driver/d4n/d4n_directory.cc

```diff
@@ -49,6 +48,14 @@ void redis_exec(std::shared_ptr<connection> conn,
   }
 }
 
+bool to_bool(std::string str) {
```

Comment:
the dirty flag is stored as a string correct? what if someone uses `update_field` to set a value of `True` or `False`. Or `TRUE` or `FALSE`. There could be multiple possibilities. You could either flag them as invalid values or handle them correctly.